activesupport/test/cache/behaviors/local_cache_behavior.rb

```diff
@@ -102,6 +102,23 @@ def test_local_cache_of_delete
     end
   end
 
+  def test_local_cache_of_delete_matched
+    begin
+      @cache.delete_matched("*")
+    rescue NotImplementedError
+      skip
+    end
+
+    @cache.with_local_cache do
+      @cache.write("foo", "bar")
+      @cache.write("fop", "bar")
+      @cache.write("bar", "foo")
+      @cache.delete_matched("fo*")
+      assert_nil @cache.read("foo")
+      assert_nil @cache.read("fop")
```

Comment:
Should we assert that non-matching keys can still be read correctly? Otherwise, why write `bar` to the cache at all?

```suggestion
      assert_nil @cache.read("fop")
      assert_equal "foo", @cache.read("bar")
```